File: 프로그래머스/unrated/181862. 세 개의 구분자/세 개의 구분자.py
def solution(myStr):

    ans = []
    for my in myStr.split('a'):
        for m in my.split('b'):
            if (m.split('c')) != '':
                ans.append(m.split('c'))
    
    ans = [a for an in ans for a in an if a != '']
    return ans if ans else ["EMPTY"]

File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.java
import java.util.*;
class Solution {
    public String[] solution(String myStr) {

        String[] split = myStr.split("a|b|c");
        List<String> filter = new ArrayList<>();
        
        for(String s: split) {
            if (!s.isEmpty()) {
                filter.add(s);
            }
        }
        String[] arr = filter.toArray(new String[filter.size()]);
        return arr.length > 0 ? arr : new String[]{"EMPTY"};
    }
}